plugins/web-terminal/src/components/TerminalComponent/TerminalComponent.tsx

```diff
@@ -128,7 +177,41 @@ export const TerminalComponent = () => {
             type="password"
             variant="outlined"
             inputRef={tokenRef}
+            required
           />
+          <Dialog open={displayModal} onClose={() => setDisplayModal(false)}>
+            <DialogTitle>Choose a namespace</DialogTitle>
+            <DialogContent>
+              <DialogContentText>
+                Unable to create devworkspace resource in {usedNamespace}{' '}
+                namespace, please choose a different one
+              </DialogContentText>
+              <form className={classes.container}>
```

Comment:
Please do not nest `form`s, that's not valid HTML.

https://html.spec.whatwg.org/multipage/forms.html#the-form-element


These are 2 different forms.

1. One for the token input + submit button.
2. A second, optional one, shown in case if the namespace is not accessible -> showing a dropdown + own submit button (it is also possible to close this modal without selecting the namespace and resume interacting with the application without Terminal being started)

Reply:
I have tried rewriting it according to you suggestions wdyt @tumido